server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesFetcher.java

```
@@ -249,12 +249,17 @@ private static boolean canMatchShard(
         QueryBuilder indexFilter,
         long nowInMillis,
         SearchExecutionContext searchExecutionContext
-    ) throws IOException {
+    ) {
         assert alwaysMatches(indexFilter) == false : "should not be called for always matching [" + indexFilter + "]";
         assert nowInMillis != 0L;
         ShardSearchRequest searchRequest = new ShardSearchRequest(shardId, nowInMillis, AliasFilter.EMPTY);
         searchRequest.source(new SearchSourceBuilder().query(indexFilter));
-        return SearchService.queryStillMatchesAfterRewrite(searchRequest, searchExecutionContext);
+        try {
+            return SearchService.queryStillMatchesAfterRewrite(searchRequest, searchExecutionContext);
+        } catch (Exception e) {
+            // treat as if shard is still a potential match
+            return true;
```

@javanna I know we discussed that logging isn't necessary here, but silently swallowing an exception is such a big anti-pattern. IMO we should at least add a WARN log entry so these events are traceable. WDYT?

---

This is fine in this case. Nobody will do anything with that log line if we do log it anyways. We do this in other places too.